File: q04/01.js
function cutbar(n, m, current) {
  if (current >= n) {
    return 0;
  } else if (current < m) {
    return 1 + cutbar(n, m, current * 2);
  } else {
    return 1 + cutbar(n, m, current + m);
  }
}

console.log(cutbar(20, 3, 1));
console.log(cutbar(100, 5, 1));

File: q04/01.swift
#!/usr/bin/swift

func cutbar(n: Int, m: Int, current: Int) -> Int {
    if current >= n {
        return 0
    }
    else if current < m {
        return 1 + cutbar(n: n, m: m, current: current * 2)
    }
    else {
        return 1 + cutbar(n: n, m: m, current: current + m)
    }
}

print(cutbar(n: 20, m: 3, current: 1))
print(cutbar(n: 100, m: 5, current: 1))
